Build fails with pnpm 11

A user reported that the current release no longer builds from the AUR after Arch Linux updated pnpm to v11.

The reported errors include:

- `ERR_PNPM_EXOTIC_SUBDEP`
- `ERR_PNPM_IGNORED_BUILDS`
- `electron-forge` reporting that `node-linker` must be hoisted

From the investigation in the AUR comments, this appears to be related to configuration changes in pnpm 11. The project currently keeps pnpm configuration in `ui/package.json` and `ui/.npmrc`, while pnpm 11 appears to expect some of these settings in `ui/pnpm-workspace.yaml`.

I'm opening this issue to track the incompatibility with pnpm 11.

--- a/ui/.npmrc
+++ b/ui/.npmrc
@@ -1,5 +1,0 @@
-registry=https://registry.npmjs.org/
-prefer-offline=true
-node-linker=hoisted
-supportedArchitectures.os=current,linux,darwin,win32
-supportedArchitectures.cpu=current,x64,arm64

--- a/ui/package.json
+++ b/ui/package.json
@@ -13,17 +13,5 @@
   },
   "devDependencies": {
     "@changesets/cli": "^2.27.1"
-  },
-  "pnpm": {
-    "overrides": {
-      "react": "^19.2.4",
-      "react-dom": "^19.2.4"
-    },
-    "onlyBuiltDependencies": [
-      "@modelcontextprotocol/ext-apps",
-      "electron",
-      "electron-winstaller",
-      "esbuild"
-    ]
   }
 }

--- a/ui/pnpm-workspace.yaml
+++ b/ui/pnpm-workspace.yaml
@@ -3,3 +3,20 @@ packages:
   - 'text'
   - 'desktop'
   - 'goose-binary/*'
+
+registry: https://registry.npmjs.org/
+preferOffline: true
+nodeLinker: hoisted
+supportedArchitectures:
+  os: [current, linux, darwin, win32]
+  cpu: [current, x64, arm64]
+overrides:
+  react: ^19.2.4
+  react-dom: ^19.2.4
+allowBuilds:
+  '@modelcontextprotocol/ext-apps': true
+  electron: true
+  electron-winstaller: true
+  esbuild: true
+# Electron Forge resolves @electron/node-gyp from git.
+blockExoticSubdeps: false

